cli/src/movement_util.rs

```diff
@@ -59,52 +59,110 @@ impl Direction {
         }
     }
 
-    fn target_not_found_message(&self, change_offset: u64) -> String {
-        match self {
-            Direction::Next => format!("No descendant found {} commit(s) forward", change_offset),
-            Direction::Prev => format!("No ancestor found {} commit(s) back", change_offset),
-        }
+    fn target_not_found_error(
+        &self,
+        workspace_command: &WorkspaceCommandHelper,
+        args: &MovementArgsInternal,
+        commits: &[Commit],
+    ) -> CommandError {
+        let err_msg = match (self, args.should_edit, args.conflict) {
+            // in edit mode, start_revset is the WC, so
+            // we only look for direct descendants.
+
+            // edit, conflict
+            (Direction::Next, true, true) => {
+                String::from("The working copy has no descendants with conflicts")
+            }
+
+            // edit, no-conflict
```

Comment:
nit: I would remove these `(no-)edit, (no-)conflict` comments and blank lines so the match source expression are placed closer.

You can also assign `args.offset` to temporary so `"{offset}"` can be inlined.

Reply:
Done.

> nit: I would remove these `(no-)edit, (no-)conflict` comments and blank lines so the match source expression are placed closer.
> 

Done.

> You can also assign `args.offset` to temporary so `"{offset}"` can be inlined.

Done.